src/components/Calculator.js

```diff
@@ -1,44 +1,60 @@
 /* eslint-disable react/prefer-stateless-function */
 import React from 'react';
 import '../App.css';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+    };
+    this.onClickHandler = this.onClickHandler.bind(this);
+  }
+
+  onClickHandler(e) {
+    this.setState((state) => calculate(state, e.target.innerHTML));
+  }
```

Comment:
**Changes Required**
- [ ] Every time a button is clicked, the state update will re-render the `virtual DOM`. To handle this in `Class Components`, we need to usually wrap this `side-effect` inside lifecycle methods like `ComponentDidMount` and `ComponentWillUpdate`. Let us try to use lifecycle methods to handle this DOM side effect.